src/convertToObject.js

```diff
@@ -12,7 +12,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+
+  const arr = sourceString.split(';').map(elem => elem.trim());
+
+  const newArr = arr.filter(char => Boolean(char));
+
+  newArr.map(item => {
+    const [key, value] = item.split(':');
+
+    obj[key.trim()] = value.trim();
+
+    return item;
+  });
```

Comment:
you're trying to do actions that are usually done in forEach

where do you return `item`? you don't assign the result anywhere